### ISSUE TITLE
Release devtools 2.4.1

### ISSUE THREAD

**Issue by jimhester** [r-lib/devtools#2348]
Prepare for release:

* [x] Check [current CRAN check results](https://cran.rstudio.org/web/checks/check_results_devtools.html)
* [x] [Polish NEWS](https://style.tidyverse.org/news.html#news-release)
* [x] [`urlchecker::url_check()`](https://github.com/r-lib/urlchecker)
* [x] `devtools::check(remote = TRUE, manual = TRUE)`
* [ ] `devtools::check_win_devel()`
* [ ] `rhub::check_for_cran()`
* [ ] `revdepcheck::revdep_check(num_workers = 4)`
* [x] Update `cran-comments.md`

Submit to CRAN:

* [x] `usethis::use_version('patch')`
* [x] `devtools::submit_cran()`
* [x] Approve email

Wait for CRAN...

* [x] Accepted :tada:
* [ ] `usethis::use_github_release()`
* [ ] `usethis::use_dev_version()`

**Comment by pingzh** [comment]
Hi @jimhester ,

Thanks for releasing this new version.

Since the new version, our docker image build is broken and we think it might be related to this new version.

This is the error:

```
                                    ^
make: *** [icu69/common/uarrsort.o] Error 1
ERROR: compilation failed for package ‘stringi’
* removing ‘/usr/local/lib/R/site-library/stringi’
ERROR: dependency ‘stringi’ is not available for package ‘stringr’
* removing ‘/usr/local/lib/R/site-library/stringr’
ERROR: dependency ‘gert’ is not available for package ‘usethis’
* removing ‘/usr/local/lib/R/site-library/usethis’
ERROR: dependency ‘stringr’ is not available for package ‘knitr’
* removing ‘/usr/local/lib/R/site-library/knitr’
ERROR: dependencies ‘knitr’, ‘stringi’, ‘stringr’ are not available for package ‘roxygen2’
* removing ‘/usr/local/lib/R/site-library/roxygen2’
ERROR: dependencies ‘usethis’, ‘roxygen2’ are not available for package ‘devtools’
* removing ‘/usr/local/lib/R/site-library/devtools’

The downloaded source packages are in
        ‘/tmp/RtmprG4mzd/downloaded_packages’
Warning messages:
1: In install.packages("devtools", repo = "http://cran.us.r-project.org") :
  installation of package ‘gert’ had non-zero exit status
2: In install.packages("devtools", repo = "http://cran.us.r-project.org") :
  installation of package ‘stringi’ had non-zero exit status
3: In install.packages("devtools", repo = "http://cran.us.r-project.org") :
  installation of package ‘stringr’ had non-zero exit status
4: In install.packages("devtools", repo = "http://cran.us.r-project.org") :
  installation of package ‘usethis’ had non-zero exit status
5: In install.packages("devtools", repo = "http://cran.us.r-project.org") :
  installation of package ‘knitr’ had non-zero exit status
6: In install.packages("devtools", repo = "http://cran.us.r-project.org") :
  installation of package ‘roxygen2’ had non-zero exit status
7: In install.packages("devtools", repo = "http://cran.us.r-project.org") :
  installation of package ‘devtools’ had non-zero exit status
>
```

OS:
```
root@5a4d4ee495a7:/# uname -a
Linux 5a4d4ee495a7 4.4.0-210-generic #242~14.04.1-Ubuntu SMP Fri Apr 16 15:01:25 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
```
R version: `R version 3.6.0 (2019-04-26)`


Some errors while building:

```
cu69/common/uarrsort.cpp: In function ‘constexpr int32_t sizeInMaxAlignTs(int32_t)’:
icu69/common/uarrsort.cpp:40:34: error: ‘max_align_t’ is not a member of ‘std’
     return (sizeInBytes + sizeof(std::max_align_t) - 1) / sizeof(std::max_align_t);
                                  ^
icu69/common/uarrsort.cpp:40:34: note: suggested alternative:
In file included from /usr/include/c++/4.8/cstddef:42:0,
                 from icu69/common/uarrsort.cpp:21:
/mnt/usr/lib/gcc/x86_64-linux-gnu/4.8/include/stddef.h:425:3: note:   ‘max_align_t’
 } max_align_t;
   ^
icu69/common/uarrsort.cpp:40:66: error: ‘max_align_t’ is not a member of ‘std’
     return (sizeInBytes + sizeof(std::max_align_t) - 1) / sizeof(std::max_align_t);
                                                                  ^
icu69/common/uarrsort.cpp:40:66: note: suggested alternative:
In file included from /usr/include/c++/4.8/cstddef:42:0,
                 from icu69/common/uarrsort.cpp:21:
/mnt/usr/lib/gcc/x86_64-linux-gnu/4.8/include/stddef.h:425:3: note:   ‘max_align_t’
 } max_align_t;
   ^
icu69/common/uarrsort.cpp:41:1: error: body of constexpr function ‘constexpr int32_t sizeInMaxAlignTs(int32_t)’ not a return-statement
 }
 ^
icu69/common/uarrsort.cpp: In function ‘void insertionSort(char*, int32_t, int32_t, int32_t (*)(const void*, const void*, const void*), const void*, UErrorCode*)’:
icu69/common/uarrsort.cpp:144:26: error: ‘max_align_t’ is not a member of ‘std’
     icu::MaybeStackArray<std::max_align_t, sizeInMaxAlignTs(STACK_ITEM_SIZE)> v;
                          ^
icu69/common/uarrsort.cpp:144:26: note: suggested alternative:
In file included from /usr/include/c++/4.8/cstddef:42:0,
                 from icu69/common/uarrsort.cpp:21:
/mnt/usr/lib/gcc/x86_64-linux-gnu/4.8/include/stddef.h:425:3: note:   ‘max_align_t’
 } max_align_t;
   ^
icu69/common/uarrsort.cpp:144:26: error: ‘max_align_t’ is not a member of ‘std’
     icu::MaybeStackArray<std::max_align_t, sizeInMaxAlignTs(STACK_ITEM_SIZE)> v;
                          ^
icu69/common/uarrsort.cpp:144:26: note: suggested alternative:
In file included from /usr/include/c++/4.8/cstddef:42:0,
                 from icu69/common/uarrsort.cpp:21:
/mnt/usr/lib/gcc/x86_64-linux-gnu/4.8/include/stddef.h:425:3: note:   ‘max_align_t’
 } max_align_t;
   ^
icu69/common/uarrsort.cpp:144:77: error: template argument 1 is invalid
     icu::MaybeStackArray<std::max_align_t, sizeInMaxAlignTs(STACK_ITEM_SIZE)> v;
                                                                             ^
icu69/common/uarrsort.cpp:144:76: error: ‘constexpr int32_t sizeInMaxAlignTs(int32_t)’ called in a constant expression
     icu::MaybeStackArray<std::max_align_t, sizeInMaxAlignTs(STACK_ITEM_SIZE)> v;
                                                                            ^
icu69/common/uarrsort.cpp:144:77: note: in template argument for type ‘int’
     icu::MaybeStackArray<std::max_align_t, sizeInMaxAlignTs(STACK_ITEM_SIZE)> v;
                                                                             ^
icu69/common/uarrsort.cpp:144:80: error: invalid type in declaration before ‘;’ token
     icu::MaybeStackArray<std::max_align_t, sizeInMaxAlignTs(STACK_ITEM_SIZE)> v;
                                                                                ^
icu69/common/uarrsort.cpp:145:40: error: request for member ‘getCapacity’ in ‘v’, which is of non-class type ‘int’
     if (sizeInMaxAlignTs(itemSize) > v.getCapacity() &&
                                        ^
icu69/common/uarrsort.cpp:146:15: error: request for member ‘resize’ in ‘v’, which is of non-class type ‘int’
             v.resize(sizeInMaxAlignTs(itemSize)) == nullptr) {
               ^
icu69/common/uarrsort.cpp:151:62: error: request for member ‘getAlias’ in ‘v’, which is of non-class type ‘int’
     doInsertionSort(array, length, itemSize, cmp, context, v.getAlias());
                                                              ^
icu69/common/uarrsort.cpp: In function ‘void quickSort(char*, int32_t, int32_t, int32_t (*)(const void*, const void*, const void*), const void*, UErrorCode*)’:
icu69/common/uarrsort.cpp:238:26: error: ‘max_align_t’ is not a member of ‘std’
     icu::MaybeStackArray<std::max_align_t, sizeInMaxAlignTs(STACK_ITEM_SIZE) * 2> xw;
                          ^
icu69/common/uarrsort.cpp:238:26: note: suggested alternative:
In file included from /usr/include/c++/4.8/cstddef:42:0,
                 from icu69/common/uarrsort.cpp:21:
/mnt/usr/lib/gcc/x86_64-linux-gnu/4.8/include/stddef.h:425:3: note:   ‘max_align_t’
 } max_align_t;
   ^
icu69/common/uarrsort.cpp:238:26: error: ‘max_align_t’ is not a member of ‘std’
     icu::MaybeStackArray<std::max_align_t, sizeInMaxAlignTs(STACK_ITEM_SIZE) * 2> xw;
                          ^
icu69/common/uarrsort.cpp:238:26: note: suggested alternative:
In file included from /usr/include/c++/4.8/cstddef:42:0,
                 from icu69/common/uarrsort.cpp:21:
/mnt/usr/lib/gcc/x86_64-linux-gnu/4.8/include/stddef.h:425:3: note:   ‘max_align_t’
 } max_align_t;
   ^
icu69/common/uarrsort.cpp:238:81: error: template argument 1 is invalid
     icu::MaybeStackArray<std::max_align_t, sizeInMaxAlignTs(STACK_ITEM_SIZE) * 2> xw;
                                                                                 ^
icu69/common/uarrsort.cpp:238:76: error: ‘constexpr int32_t sizeInMaxAlignTs(int32_t)’ called in a constant expression
     icu::MaybeStackArray<std::max_align_t, sizeInMaxAlignTs(STACK_ITEM_SIZE) * 2> xw;
                                                                            ^
icu69/common/uarrsort.cpp:238:81: note: in template argument for type ‘int’
     icu::MaybeStackArray<std::max_align_t, sizeInMaxAlignTs(STACK_ITEM_SIZE) * 2> xw;
                                                                                 ^
icu69/common/uarrsort.cpp:238:85: error: invalid type in declaration before ‘;’ token
     icu::MaybeStackArray<std::max_align_t, sizeInMaxAlignTs(STACK_ITEM_SIZE) * 2> xw;
                                                                                     ^
icu69/common/uarrsort.cpp:239:42: error: request for member ‘getCapacity’ in ‘xw’, which is of non-class type ‘int’
     if(sizeInMaxAlignTs(itemSize)*2 > xw.getCapacity() &&
                                          ^
icu69/common/uarrsort.cpp:240:16: error: request for member ‘resize’ in ‘xw’, which is of non-class type ‘int’
             xw.resize(sizeInMaxAlignTs(itemSize) * 2) == nullptr) {
                ^
icu69/common/uarrsort.cpp:246:21: error: request for member ‘getAlias’ in ‘xw’, which is of non-class type ‘int’
                  xw.getAlias(), xw.getAlias() + sizeInMaxAlignTs(itemSize));
                     ^
icu69/common/uarrsort.cpp:246:36: error: request for member ‘getAlias’ in ‘xw’, which is of non-class type ‘int’
                  xw.getAlias(), xw.getAlias() + sizeInMaxAlignTs(itemSize));
                                    ^
make: *** [icu69/common/uarrsort.o] Error 1
ERROR: compilation failed for package ‘stringi’
```

```
/usr/bin/ld: /tmp/Rtmpz5u4Or/R.INSTALL8d1a1db84c10/gert/libgit2/lib/libgit2.a(annotated_commit.c.o): unrecognized relocation (0x2a) in section `.text'
/usr/bin/ld: final link failed: Bad value
collect2: error: ld returned 1 exit status
make: *** [gert.so] Error 1
ERROR: compilation failed for package ‘gert’
```

**Comment by jimhester** [comment]
These are errors in dependencies, specifically the errors you show are coming from the stringi package. Not anything that could be fixed in devtools.

In general configuration issues are better fit for <https://community.rstudio.com>. Do you mind asking it over there?
